### PR TITLE
bazel: add tracing libraries lint to nogo

### PR DESCRIPTION
**Pull request by Strum355**
Closes https://github.com/sourcegraph/sourcegraph/issues/54838

## Test plan

Added `github.com/opentracing/opentracing-go` to some package and did `bazel build //<pkg>` to observe:
`cmd/cody-gateway/internal/actor/limits.go:15:2: import 'github.com/opentracing/opentracing-go/log' is not allowed from list 'tracing libraries': use "go.opentelemetry.io/otel/trace" instead (depguard)`

### CODE REVIEW

**Comment by Strum355** [comment]
@jhchabran @burmudar do we want to remove them from `sg lint` in the same PR?

**Comment by jhchabran** [comment]
@Strum355 yes, let's remove it as well :)

File: dev/linters/tracinglibraries/tracinglibraries.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"github.com/sourcegraph/sourcegraph/dev/linters/nolint"
+)
+
+const deniedMessage = `use "go.opentelemetry.io/otel/trace" instead`
+
+var allowedFiles = []string{
+	// Banned imports will match on the linter here
+	"!**/dev/sg/linters",
+	// Adapters here
+	"!**/internal/tracer",
```
Comment (jhchabran) [inline dev/linters/tracinglibraries/tracinglibraries.go:20]: <3 comments

**Review by jhchabran — COMMENTED** [review]
Nice :) 

Do you plan to put the `onlyLocal(lintTracingLibraries()` in another PR?

**Comment by Strum355** [comment]
> Nice :)
> 
> Do you plan to put the `onlyLocal(lintTracingLibraries()` in another PR?

done in this one!

**Review by jhchabran — APPROVED** [review]

**Pull request merged**